### PR TITLE
#27500 Prepare Newsletter module Tests for PHPUnit 8

### PR DESCRIPTION
**Pull request by lbajsarowicz**
### Description (*)
Preparing PHPUnit Tests to run with PHPUnit 8.
PRs for each module is separated to make tests easier to merge.

### Fixed Issues (if relevant)
1. magento/magento2#27500

### Manual testing scenarios (*)
N/A

### Questions or comments
PR #27701 needs to be merged first (otherwise PHPUnit tests fail)

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by slavvka** [comment]
```
18:21:12 1) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "missing_previous_and_new_status" (null, null, null)
18:21:12 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
18:21:12 
18:21:12 2) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "missing_previous_status_and_subscribe" (null, true, true)
18:21:12 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
18:21:12 
18:21:12 3) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "new_unsubscribed_value_and_missing_previous_status" (null, false, null)
18:21:12 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
18:21:12 
18:21:12 4) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "previous_subscribed_status_without_new_value" (1, null, null)
18:21:12 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
18:21:12 
18:21:12 5) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "same_subscribed_previous_and_new_status" (1, true, null)
18:21:12 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
18:21:12 
18:21:12 6) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "unsubscribe_previously_subscribed_customer" (1, false, false)
18:21:12 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
18:21:12 
18:21:12 7) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "previously_unsubscribed_status_without_new_value" (3, null, null)
18:21:12 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
18:21:12 
18:21:12 8) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "subscribe_previously_unsubscribed_customer" (3, true, true)
18:21:12 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
18:21:12 
18:21:12 9) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "same_unsubscribed_previous_and_new_status" (3, false, null)
18:21:12 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
18:21:12 
18:21:12 10) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "previous_unconfirmed_status_without_new_value" (4, null, true)
18:21:12 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
18:21:12 
18:21:12 11) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "subscribe_previously_unconfirmed_status" (4, true, true)
18:21:12 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
18:21:12 
18:21:12 12) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "unsubscribe_previously_unconfirmed_status" (4, false, true)
18:21:12 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
18:21:12 
18:21:12 13) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterGetByIdCreatesExtensionAttributes
18:21:12 Trying to configure method "setIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
```

File: app/code/Magento/Newsletter/Test/Unit/Model/Queue/TransportBuilderTest.php [hunk 1]
```diff
@@ -46,32 +44,32 @@ class TransportBuilderTest extends TestCase
     protected $builder;
 
     /**
-     * @var FactoryInterface|PHPUnit_Framework_MockObject_MockObject
+     * @var FactoryInterface|\PHPUnit\Framework\MockObject\MockObject
```
Comment (ihor-sviziev) [inline app/code/Magento/Newsletter/Test/Unit/Model/Queue/TransportBuilderTest.php:47]: Please import this class

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Hi @lbajsarowicz ,
Could you also review test failures that @slavvka added above?

**Comment by slavvka** [comment]
@lbajsarowicz I merged this PR with red Unit tests because I thought they were caused by flaky JS tests. But there is another issue with it because currently `phpunut8` branch fails with:
```
18:43:39 1) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "missing_previous_and_new_status" (null, null, null)
18:43:39 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
18:43:39 
18:43:39 2) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "missing_previous_status_and_subscribe" (null, true, true)
18:43:39 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
18:43:39 
18:43:39 3) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "new_unsubscribed_value_and_missing_previous_status" (null, false, null)
18:43:39 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
18:43:39 
18:43:39 4) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "previous_subscribed_status_without_new_value" (1, null, null)
18:43:39 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
18:43:39 
18:43:39 5) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "same_subscribed_previous_and_new_status" (1, true, null)
18:43:39 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
18:43:39 
18:43:39 6) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "unsubscribe_previously_subscribed_customer" (1, false, false)
18:43:39 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
18:43:39 
18:43:39 7) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "previously_unsubscribed_status_without_new_value" (3, null, null)
18:43:39 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
18:43:39 
18:43:39 8) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "subscribe_previously_unsubscribed_customer" (3, true, true)
18:43:39 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
18:43:39 
18:43:39 9) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "same_unsubscribed_previous_and_new_status" (3, false, null)
18:43:39 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
18:43:39 
18:43:39 10) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "previous_unconfirmed_status_without_new_value" (4, null, true)
18:43:39 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
18:43:39 
18:43:39 11) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "subscribe_previously_unconfirmed_status" (4, true, true)
18:43:39 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
18:43:39 
18:43:39 12) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterSave with data set "unsubscribe_previously_unconfirmed_status" (4, false, true)
18:43:39 Trying to configure method "getIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
18:43:39 
18:43:39 13) Magento\Newsletter\Test\Unit\Model\Plugin\CustomerPluginTest::testAfterGetByIdCreatesExtensionAttributes
18:43:39 Trying to configure method "setIsSubscribed" which cannot be configured because it does not exist, has not been specified, is final, or is static
```
Could you please also address it?

**Comment by slavvka** [comment]
Unfortunately Allure reports don't support PHPUnit warnings now whereas PHPUnit started to return exit code 1 when there are warnings. It didn't previously. I created a PR https://github.com/allure-framework/allure-phpunit/pull/48 to address that but it is still not merged whereas we cannot upgrade to Allure latest version until we upgrade PHPUnit :)

**Pull request merged**